### PR TITLE
feat(jiva-csi): Add JivaVolumePolicy schema changes

### PR DESCRIPTION
**Pull request by shubham14bajpai**
Signed-off-by: shubham shubham.bajpai@mayadata.io
This PR adds changes corresponding to the schema changes made to JivaVolume and JivaVolumePolicy CRDs. This PR also updates the target path in the JivaVolume spec. The policy name for the volume will be passed as an annotation to the JivaVolume CR.
Refer: https://github.com/openebs/jiva-operator/pull/8

### CODE REVIEW

File: .travis.yml
Comment (utkarshmani1997) [inline .travis.yml]: Please remove this once jiva-operator is merged

File: go.mod
Comment (utkarshmani1997) [inline go.mod]: same comment as above

File: pkg/driver/node.go
Comment (utkarshmani1997) [inline pkg/driver/node.go]: use doesVolumeExist function.

File: pkg/kubernetes/client/client.go
Comment (utkarshmani1997) [inline pkg/kubernetes/client/client.go]: should we create default policies in this case ?

File: pkg/kubernetes/client/client.go
Comment (utkarshmani1997) [inline pkg/kubernetes/client/client.go]: Should we move this to another function ?

File: pkg/kubernetes/client/client.go
Comment (utkarshmani1997) [inline pkg/kubernetes/client/client.go]: same comment as above

**Review by utkarshmani1997 — CHANGES_REQUESTED** [review]
@shubham14bajpai i have given few minor comments, let's discuss on the same.

File: ci/ci.sh
Comment (utkarshmani1997) [inline ci/ci.sh:14]: Please add one more test case to test the same without policy

File: pkg/jivavolume/jivavolume.go
Comment (utkarshmani1997) [inline pkg/jivavolume/jivavolume.go]: move this to another function `getDefaultTargetSpec`

File: pkg/jivavolume/jivavolume.go
Comment (utkarshmani1997) [inline pkg/jivavolume/jivavolume.go]: same comment as above

File: pkg/jivavolume/jivavolume.go
Comment (utkarshmani1997) [inline pkg/jivavolume/jivavolume.go]: Optional: can we mv entire if cases to a map and then iterate over it to set default values ?

File: pkg/jivavolume/jivavolume.go
Comment (utkarshmani1997) [inline pkg/jivavolume/jivavolume.go]: same comment as above

**Review by utkarshmani1997 — CHANGES_REQUESTED** [review]
@shubham14bajpai i have provided few minor comments, please have a look at it. I will approve this once jiva-operator PR is merged.

File: ci/ci.sh
Comment (shubham14bajpai) [reply ci/ci.sh:14]: Without the policy, the default replicationfactor will be 3 which cannot be tested on a single node.

**Review by utkarshmani1997 — APPROVED** [review]
LGTM

File: go.mod
Comment (prateekpandey14) [inline go.mod]: we have to remove this change ? once the operator changes merged

**Review by prateekpandey14 — APPROVED** [review]
some minor not picks and cleanups  deps , otherwise LGTM

**Pull request merged**
